### PR TITLE
refactor(docs): Updating GraphQL documentation with fixes, and more examples

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Summary**

In this PR, we make major improvements to GraphQL documentation, including fixing erroneous & confusing legacy documentation, and adding a ton of new examples on how to read, write, and search metadata.

**Status**

Ready for review

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/api/graphql/getting-started.md [hunk 1]
```diff
@@ -17,54 +17,40 @@ We'll reiterate a few important points before proceeding:
...
-   
-> Important: The DataHub GraphQL endpoint only supports POST requests at this time. It does not support GET requests. If this is something
-> you need, let us know on Slack!
+> Notice: The DataHub GraphQL endpoint only supports POST requests at this time.
 
 ### On the Horizon
 
-- **Service Access Tokens**: In the near future, the DataHub team intends to introduce service users, which will provide a way to generate and use API access
+- **Service Tokens**: In the near future, the DataHub team intends to introduce service users, which will provide a way to generate and use API access
 tokens when querying both the Frontend Proxy Server and the Metadata Service. If you're interested in contributing, please [reach out on our Slack](https://datahubspace.slack.com/join/shared_invite/zt-nx7i0dj7-I3IJYC551vpnvvjIaNRRGw#/shared-invite/email).
```
Comment (shirshanka) [inline docs/api/graphql/getting-started.md:45]: this link may not be valid. I think `https://slack.datahubproject.io` is the evergreen link.

Comment (jjoyce0510) [reply docs/api/graphql/getting-started.md:45]: Updated!

File: docs/api/graphql/getting-started.md [hunk 2]
```diff
@@ -17,62 +17,48 @@ We'll reiterate a few important points before proceeding:
...
+In general, you'll need to provide an [Access Token](../../authentication/personal-access-tokens.md) when querying the GraphQL by
+providing an `Authorization` header containing a `Bearer` token. The header should take the following format:
 
-There are a few options when it comes to querying the GraphQL endpoint. The recommendation on which to use varies by use case.
+```bash
+Authorization: Bearer <access-token>
+```
 
-**Testing**: [Postman](https://learning.postman.com/docs/sending-requests/supported-api-frameworks/graphql/), GraphQL Explorer (described below), CURL
+Authorization for actions exposed by the GraphQL endpoint will be performed based  on the actor making the request.
```
Comment (chriscollins3456) [inline docs/api/graphql/getting-started.md:37]: super nit - there are two spaces after "based" before "on"

File: docs/api/graphql/getting-started.md [hunk 3]
```diff
@@ -17,62 +17,48 @@ We'll reiterate a few important points before proceeding:
...
 - **DataHub Client SDKs**: Libraries wrapping the DataHub GraphQL API on a per-language basis (based on community demand). 
 
 ## GraphQL Explorer 
 
-DataHub provides a browser-based GraphQL Explorer Tool ([GraphiQL](https://github.com/graphql/graphiql)) for live interaction with the GraphQL API. Today, this tool is available for use in multiple places (like the GraphQL endpoint itself):
-
-1. **Metadata Service**: `http://<metadata-service-address>/api/graphiql`. For local deployments, `http://localhost:8080/api/graphiql`.
-2. **Frontend Proxy**: `http://<frontend-service-address>/api/graphiql`. For local deployments, `http://localhost:9002/api/graphiql`.
-
+DataHub provides a browser-based GraphQL Explorer Tool ([GraphiQL](https://github.com/graphql/graphiql)) for live interaction with the GraphQL API. This tool is available at the path `/api/v2/graphiql` (e.g. `https://my-company.datahub.com/api/v2/graphql`)
```
Comment (chriscollins3456) [inline docs/api/graphql/getting-started.md:50]: oh there's a `v2` now? also I think you need to end the example link with `graphiql` not just `graphql`

**Review by chriscollins3456 — COMMENTED** [review]
nice stuff! I think this will really help a lot of people in the community

File: docs/api/graphql/getting-started.md (hunk 3 above)
Comment (jjoyce0510) [reply docs/api/graphql/getting-started.md:50]: Yes good catch! Actually v2 is an old thing we can do the normal one

**Review by chriscollins3456 — APPROVED** [review]
nice rice!

**Pull request merged**